### DISCUSSION TITLE
Where to find examples of using API to update objects?

### DISCUSSION THREAD

**Discussion by lonelyadmin** [netbox-community/netbox#5850] [Q&A] [answered]
I need to update massive amounts of front_ports. If the import CSV would do updates for existing objects I'd be good, but it looks like I need to use the API for this. Are there any examples around of scripts that do this?

And yes, I did RTFM...I just need some examples to go by. Is pynetbox the way to go on this?

**Comment by markkuleinio** [comment]
I haven't worked with ports with pynetbox but here are some examples of using it anyway:

https://majornetwork.net/2020/06/using-pynetbox/

**Comment by MrPaulAR** [comment]
This is a snippit from the code I use to deploy ports.  Granted, I'm lazilly doing this one port at a time and, at least for me, pynetbox seems to be the way to go.

```
nb_device = nb.dcim.devices.get(name__ic=HOSTNAME)
nb.dcim.interfaces.create(
        device=nb_device.id,
        name="ge-0/0/0",
        type="10gbase-x-sfpp",
    )
```

**Comment by DCAuto** [comment, marked as answer]
pynetbox is for sure the way to go for updating. It's quick and easy. 

1) Query the object you want to update
```
        nb_interface = nb.dcim.interfaces.get(
            name=interface,
            device_id=nb_device.id
        )
```
2) Update fields you want to (I think there's an easier way to do this part but this works so I never looked)
```            tagged_vlans = []
            new_vlan_id = 10
           # add existing vlans and new vlan to a list
           for vlan in nb_interface.tagged_vlans:
                tagged_vlans.append(vlan.id)
           tagged_vlans.append(new_vlan_id)
            nb_interface.tagged_vlans = tagged_vlans
```

3) Save the object
`            nb_interface.save()`
